***************
*** 19,30 ****
  class A {
  }
  class TypeChanges {
!   class func method(with a: A?) -> A
!   func method(with a: A?) -> A
!   @available(swift, obsoleted: 3, renamed: "method(with:)")
!   class func methodWithA(_ a: A?) -> A
!   @available(swift, obsoleted: 3, renamed: "method(with:)")
!   func methodWithA(_ a: A?) -> A
  }
  class Base {
    init()
--- 19,30 ----
  class A {
  }
  class TypeChanges {
!   class func method(withA a: Any) -> Any
!   func method(withA a: Any) -> Any
!   @available(swift, obsoleted: 3, renamed: "method(withA:)")
!   class func methodWithA(_ a: Any) -> Any
!   @available(swift, obsoleted: 3, renamed: "method(withA:)")
!   func methodWithA(_ a: Any) -> Any
  }
  class Base {
    init()
***************
*** 318,325 ****
    init()
  }
  extension TestProperties {
!   func accessorsOnlyWithNewType() -> Base
!   func setAccessorsOnlyWithNewType(_ accessorsOnlyWithNewType: Base)
  }
  extension TestProperties {
    var accessorsOnlyRenamedRetyped: Any!
--- 318,325 ----
    init()
  }
  extension TestProperties {
!   func accessorsOnlyWithNewType() -> Any
!   func setAccessorsOnlyWithNewType(_ accessorsOnlyWithNewType: Any)
  }
  extension TestProperties {
    var accessorsOnlyRenamedRetyped: Any!
